src/static_handler.rs

```diff
@@ -5,6 +5,8 @@ use std::error::Error;
 use std::fmt;
 use time::{self, Timespec};
 
+use std::os::unix::prelude::MetadataExt;
```

Comment:
Yes unfortunately there is not currently a stable method of acquiring the modification time of a file in a cross-platform fashion. Note that the code before was also not correctly cross-platform because Windows timestamps are relative to a different point in the past than Unix timestamps.

To learn about the modification time of a file (without caring about the value) in a cross platform fashion I would recommend something along the lines of [Cargo's `MTime`](https://github.com/rust-lang/cargo/blob/master/src/cargo/util/mtime.rs), and to make sure the two times are offset to the right value I would recommend just frobbing the returned value on windows [with a particular constant](https://github.com/alexcrichton/tar-rs/blob/master/src/lib.rs#L463-L467)